### PR TITLE
CAL-227 Increased FMV stream shutdown speed

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
This PR reduces the time it takes to close netty channels when the stream is shut down.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@glenhein @rzwiefel 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire
#### How should this be tested?
Build / Install Experimental / Stream Video / Shut down stream during transmission / Observe shutdown performance.  No new content should be created after a few seconds.
#### Any background context you want to provide?
Prior to this fix, the netty channel would continue to receive data after shutdown was called.
#### What are the relevant tickets?
[CAL-227](https://codice.atlassian.net/browse/CAL-227)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java [hunk 1]
```diff
@@ -432,8 +445,6 @@ public void updateCallback(Map<String, Object> properties) {
                 return;
             }
 
-            setParentTitle((String) properties.get(METATYPE_TITLE));
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: This line occurs twice in the update method

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java [hunk 2]
```diff
@@ -129,7 +130,7 @@
      */
     private static final String METATYPE_PARENT_TITLE = "parentTitle";
 
-    private static final String INTERFACE = "interface";
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: This string is not used

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java [hunk 3]
```diff
@@ -363,7 +364,17 @@ public void destroy(int arg) {
...
-            eventLoopGroup.shutdownGracefully();
+            try {
+                eventLoopGroup.shutdownGracefully()
+                        .sync();
+                if (channelFuture != null) {
+                    channelFuture.channel()
+                            .closeFuture()
+                            .sync();
+                }
+            } catch (InterruptedException e) {
```
Comment (jlcsmith) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: move the closeFuture() to a finally block?

**Comment by jlcsmith** [comment]
@bdeining - can you characterize how quickly the stream shuts down once it is requested?

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java (hunk 3 above)
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: That may clutter up some of this code ; both `sync()`s throw an InterruptedException (we'd have to catch it in the finally block)

**Comment by bdeining** [comment]
@jlcsmith the stream should shutdown in less than 5 seconds

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java (hunk 3 above)
Comment (kcwire) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: If the first sync throws an `InterruptedException` what state does that leave the `channelFuture`? We need to ensure both on closed/shutdowned and synced.

**Comment by kcwire** [comment]
Tested and confirmed the stream shuts down in a timely manner on demand of the administrator.

**Pull request merged**